packs/elastic/actions/lib/utils.py

```diff
@@ -0,0 +1,68 @@
+import elasticsearch
+import curator
+import sys
+import logging
+from curator.api.utils import get_version, is_master_node
+
+# Elasticsearch versions supported
+version_max  = (2, 0, 0)
+version_min = (1, 0, 0)
+logger = logging.getLogger(__name__)
+
+ES_ARGS = ['host', 'url_prefix', 'port', 'http_auth', 'use_ssl']
+
+
+def check_version(client):
+    """
+    Verify version is within acceptable range.  Exit with error if it is not.
+    :arg client: The Elasticsearch client connection
+    """
+    version_number = get_version(client)
+    logger.debug('Detected Elasticsearch version {0}'.format(".".join(map(str,version_number))))
+    if version_number >= version_max or version_number < version_min:
+        print 'Expected Elasticsearch version range > {0} < {1}'.format(".".join(map(str,version_min)),".".join(map(str,version_max)))
+        print 'ERROR: Incompatible with version {0} of Elasticsearch.  Exiting.'.format(".".join(map(str,version_number)))
+        sys.exit(1)
+
+
+def check_master(client, master_only=False):
+    """
+    Check if master node.  If not, exit with error code
+    """
+    if master_only and not is_master_node(client):
+        logger.info('Master-only flag detected. Connected to non-master node. Aborting.')
+        sys.exit(9)
+
+
+def get_client(**kwargs):
```

Comment:
So in this case we could improve on that and at least make it more readable in our pack :)